### PR TITLE
gballoc hl win32

### PR DESCRIPTION
**Pull request by anporumb**

### CODE REVIEW

File: linux/tests/gballoc_hl_passthrough_ut/gballoc_hl_passthrough_ut.c [hunk 1]
```diff
@@ -180,7 +180,7 @@ TEST_FUNCTION(gballoc_hl_malloc_unhappy_path)
     ///clean
 }
 
-/*Tests_SRS_GBALLOC_HL_PASSTHROUGH_02_006: [ gballoc_hl_free shall call gballoc_hl_free(ptr). ]*/
```
Comment (dcristoloveanu) [inline linux/tests/gballoc_hl_passthrough_ut/gballoc_hl_passthrough_ut.c:183]: >h [](http://example.com/codeflow?start=80&length=1)

lol

File: win32/devdoc/gballoc_hl_passthrough.md [hunk 2]
```diff
@@ -107,14 +111,41 @@ MOCKABLE_FUNCTION(, void*, gballoc_hl_malloc, size_t, size);
...
+**SRS_GBALLOC_HL_PASSTHROUGH_02_032: [** If `lazy_init` fail then `gballoc_hl_malloc_flex` shall fail and return `NULL`. **]**
+
+**SRS_GBALLOC_HL_PASSTHROUGH_02_033: [** `gballoc_hl_malloc_flex` shall call `gballoc_ll_malloc_flex(size)` and return what `gballoc_hl_malloc_flex` returned. **]**
+
 ### gballoc_hl_free
 ```c
 MOCKABLE_FUNCTION(, void, gballoc_hl_free, void*, ptr);
 ```
 
 `gballoc_hl_free` calls `gballoc_hl_free(ptr)`.
```
Comment (dcristoloveanu) [inline win32/devdoc/gballoc_hl_passthrough.md:146]: >lloc_hl_free` calls `gballoc_hl_free(ptr)`. [](http://example.com/codeflow?start=4&length=43)

inception!

Comment (anporumb) [reply win32/devdoc/gballoc_hl_passthrough.md:146]: un-inception'd

File: win32/tests/gballoc_hl_metrics_ut/gballoc_hl_metrics_ut.c [hunk 3]
```diff
@@ -321,6 +322,175 @@ TEST_FUNCTION(gballoc_hl_malloc_when_not_initialized_returns_NULL)
...
+    ///arrange
+    void* result;
+    void* gballoc_ll_result;
+    STRICT_EXPECTED_CALL(gballoc_ll_init(NULL));
+    (void)gballoc_hl_init(NULL, NULL);
+    umock_c_reset_all_calls();
+
+    STRICT_EXPECTED_CALL(lazy_init(IGNORED_ARG, IGNORED_ARG, NULL));
+    STRICT_EXPECTED_CALL(timer_global_get_elapsed_us());
+    STRICT_EXPECTED_CALL(gballoc_ll_malloc_flex(2,3,5))
```
[the remark this answers is not in the document]
Comment (anporumb) [reply win32/tests/gballoc_hl_metrics_ut/gballoc_hl_metrics_ut.c:424]: Testing with 0 at this layer doesn't bring anything, arguments pass "as they are" to _ll layer where they are read and something happens based on the value of the arguments (including a /div 0 check).

File: win32/src/gballoc_hl_metrics.c [hunk 4]
```diff
@@ -346,6 +346,74 @@ void* gballoc_hl_malloc(size_t size)
...
+
+        if (result == NULL)
+        {
+            LogError("failure in gballoc_ll_malloc_flex(base=%zu, nmemb=%zu, size=%zu)", base, nmemb, size);
+        }
+
+        /*Codes_SRS_GBALLOC_HL_METRICS_02_011: [ gballoc_hl_malloc_flex shall call timer_global_get_elapsed_us to obtain the end time of the allocate. ]*/
+        double end_time = timer_global_get_elapsed_us();
+
+        LONG latency = (LONG)(end_time - start_time);
```
[the remark this answers is not in the document]
Comment (anporumb) [reply win32/src/gballoc_hl_metrics.c:410]: Task created: Task 12357398: eliminate LONG from gballoc_hl.

**Pull request merged**